### PR TITLE
Strict null Cleanup

### PR DESCRIPTION
**Pull request by blink1073**
Used `strictNullChecks` to clean up `coreutils` and `services`.  We could not leave the check on in `tsconfig` yet because `JSONObject` cannot yet take an optional key.

This effort caught some bugs where we were not properly handling `null` or `undefined` and helped clarify the `services` interfaces.

### CODE REVIEW

File: packages/services/src/serverconnection.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ namespace ServerConnection {
   function makeSettings(options?: Partial<ISettings>) {
     // Use the singleton default settings if no options are given.
     if (options === void 0) {
-      if (Private.defaultSettings === null) {
+      if (Private.defaultSettings === void 0) {
```
Comment (afshin) [inline packages/services/src/serverconnection.ts]: Should this be `undefined`?

Comment (blink1073) [inline packages/services/src/serverconnection.ts]: They're equivalent, and this is internal.

Comment (afshin) [inline packages/services/src/serverconnection.ts]: 👍

**Comment by vidartf** [comment]
> JSONObject cannot yet take an optional key.

Could this be fixed by (temporarily?) using `skipLibCheck` compiler option?

**Comment by blink1073** [comment]
True, but I'd rather not use that hammer even temporarily.

**Pull request merged**